### PR TITLE
Mayhem integration using Github Action

### PR DESCRIPTION
**Pull request by Verylostpenguin**

### CODE REVIEW

File: Mayhemfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+project: Verylostpenguin/rav1e
+target: rav1e
+duration: 90
+
+cmds:
+  - cmd: /rav1e/fuzz/target/x86_64-unknown-linux-gnu/release/encode_decode
```
Comment (nathanjackson) [inline Mayhemfile:6]: We need a few changes to your Mayhemfile:

- Remove your username from the project.
- No need to include duration.
- Please remove all but one `cmd` and rename the `target` field to reflect it. Mayhemfiles should be written for one target, but the multiple commands listed here would actually be for multiple targets.

For example, your Mayhemfile should look something like this:

```
project: rav1e
target: encode_decode

cmds:
  - cmd: /rav1e/fuzz/target/x86_64-unknown-linux-gnu/release/encode_decode
```

**Review by nathanjackson — CHANGES_REQUESTED** [review]
Thanks for your PR.

See my comments, there are a few things we need to have fixed before we can merge this.

File: Mayhemfile (hunk 1 above)
Comment (Verylostpenguin) [reply Mayhemfile]: I've deleted my username and the duration per the first two points.

I've split the commands in the mayhem files into multiple mayhemfiles that are now all called sequentially in mayhem.yml.

File: .github/workflows/mayhem.yml [hunk 2]
```diff
@@ -1,51 +1,88 @@
...
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Start analysis for encode
+        uses: ForAllSecure/mcode-action@v1
+        with:
+          mayhem-token: ${{ secrets.MAYHEM_TOKEN }}
+          args: --image ${{ steps.meta.outputs.tags }}
+                --file ${{ env.ENCODE_MAYHEMFILE }}
```
Comment (nathanjackson) [inline .github/workflows/mayhem.yml:58]: I believe you need to prepend `mayhemfiles` to the file argument. See your other workflow: https://github.com/mayhemheroes/smoltcp/blob/master/.github/workflows/mayhem.yml

**Review by nathanjackson — CHANGES_REQUESTED** [review]
Hey @Verylostpenguin , sorry to take so long to get back with you. I merged your other PR, but found a few issues that I myself wasn't sure how to handle. So I had to fix a couple of things to better support multiple targets.

Your latest changes for this one also look good, but you need to prepend the `mayhemfiles` directory to the `--file` argument for each mcode-action. You should be able to see what I did to smoltcp to make this work.

File: .github/workflows/mayhem.yml (hunk 2 above)
Comment (Verylostpenguin) [reply .github/workflows/mayhem.yml]: I've prepended the folder name and pushed the commit. Is there anything else that needs to be done?

Comment (nathanjackson) [reply .github/workflows/mayhem.yml]: Nope. Looks great 😄

**Review by nathanjackson — APPROVED** [review]
LGTM

**Pull request merged**